src/main/kotlin/mate/academy/Main.kt

```diff
@@ -1,7 +1,13 @@
 package mate.academy
 
 fun greetUsers() {
-    // implement function
+    println("Please enter a name (or press Enter to quit):")
+    val name = readLine() ?: ""
+    if (name.isEmpty()) {
+        return
+    } else
+    println("Hello, $name!")
+    greetUsers()
```

Comment:
Pay attention to have certain exit condition when using recursion in similar cases. Just FYI